### PR TITLE
Do not provision the stop workspace role if there is no OS OAuth identitty provider

### PR DESCRIPTION
**Pull request by tomgeorge**
### What does this PR do?

Do not provision the stop workspace role if there is no OS OAuth identitty provider configured.

Signed-off-by: Tom George <tgeorge@redhat.com>

### What issues does this PR fix or reference?

This is a more long-term fix for https://github.com/eclipse/che/issues/16724

### CODE REVIEW

**Comment by ibuziuk** [comment]
[ci-test]

**Comment by ibuziuk** [comment]
[ci-build]

File: infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java [hunk 1]
```diff
@@ -21,13 +21,7 @@
...
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.lenient;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (ibuziuk) [inline infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java]: IMO, better to use precise imports

**Review by ibuziuk — APPROVED** [review]

File: infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java (hunk 1 above)
Comment (tomgeorge) [reply infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java]: is this something `mvn fmt:format` touches?

Comment (ibuziuk) [reply infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java]: I do not think so...

Comment (tomgeorge) [reply infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java]: ok that probably happened when I organized my imports.  I will change it back.

**Review by amisevsk — APPROVED** [review]
LGTM

**Comment by ibuziuk** [comment]
[ci-build]

**Pull request merged**